lib/CL/devices/level0/pocl-level0.cc

```diff
@@ -897,34 +1038,144 @@ int pocl_level0_setup_metadata(cl_device_id Device, cl_program Program,
   return 1;
 }
 
-int pocl_level0_create_kernel(cl_device_id Device, cl_program Program,
-                              cl_kernel Kernel, unsigned ProgramDeviceI) {
-  assert(Program->data[ProgramDeviceI] != nullptr);
-  Level0Program *L0Program = (Level0Program *)Program->data[ProgramDeviceI];
-  Level0Kernel *Ker =
-      DriverInstance->getJobSched().createKernel(L0Program, Kernel->name);
-  Kernel->data[ProgramDeviceI] = Ker;
-  return Ker ? CL_SUCCESS : CL_OUT_OF_RESOURCES;
+
+#ifdef ENABLE_NPU
+
+static bool pocl_npu_is_layout_gemm(cl_uint Rank, const void *Layout) {
+  const cl_tensor_layout_ml_exp *Ptr = (cl_tensor_layout_ml_exp *)Layout;
+
+  // supported layouts from openvino compiler plugin / "rankToLegacyLayoutString":
+  // C, NC, CHW, NCHW, NCDHW
+  if (Ptr->ml_type == CL_TENSOR_LAYOUT_ML_NC_EXP && Rank == 2)
+    return true;
+  if (Ptr->ml_type == CL_TENSOR_LAYOUT_ML_CHW_EXP && Rank == 3)
+    return true;
+  return false;
 }
 
-int pocl_level0_free_kernel(cl_device_id Device, cl_program Program,
-                            cl_kernel Kernel, unsigned ProgramDeviceI) {
+int
+pocl_npu_validate_khr_gemm (cl_bool TransA,
+                            cl_bool TransB,
+                            const cl_tensor_desc_exp *TenA,
+                            const cl_tensor_desc_exp *TenB,
+                            const cl_tensor_desc_exp *TenCIOpt,
+                            const cl_tensor_desc_exp *TenCOut,
+                            const cl_tensor_datatype_value_exp *Alpha,
+                            const cl_tensor_datatype_value_exp *Beta)
+{
```

Comment:
Hm. This seems to be a .cc source. I wonder why formatter treats it as C?

Reply:
Should be fixed now (ran format-branch).